src/pocketdb/consensus/social/User.hpp

```diff
@@ -189,6 +173,21 @@ namespace PocketConsensus
 
             return Success;
         }
+    
+        virtual ConsensusValidateResult CheckLogin(const UserRef& ptx)
+        {
+            auto name = *ptx->GetPayloadName();
+
+            // Trim spaces
+            if (boost::algorithm::ends_with(name, "%20") || boost::algorithm::starts_with(name, "%20"))
```

Comment:
Why not using same logic as for `UserConsensus_checkpoint_login_limitation` but with checkpoint? Also i think checking for emptiness is reuired

Reply:
Isn't there also an issue with other non-visible characters such as %00 being included at the beginning and end of the string?

Reply:
It appears every character from 0x00 to 0x20 is a control character and should be excluded

Reply:
::isalnum should identify any bad character
But I'm not sure about the control characters - from 0x00 to 0x20

Reply:
It ok because allows only a-z, A-Z and 0-9